pkg/core/db/pool.go

```diff
@@ -0,0 +1,71 @@
+package db
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/skygeario/skygear-server/pkg/core/config"
+)
+
+type Pool interface {
+	Open(tConfig config.TenantConfiguration) (*sqlx.DB, error)
+	Dispose() error
```

Comment:
Why not just call this `Close() error`?

Reply:
No particular reason. If you prefer to name it `Close` I'll change it.